### ISSUE TITLE
There was a network error. ( file:///android_asset/www/#/status)

### ISSUE THREAD

**Issue by MatthewPringle** [isleofcode/corber#553]
The app works on Android fine on first boot.

If the app is closed and then reopened when a user has logged in then the internal url /#/status.

The app then refuses to load with the message There was a network error. ( file:///android_asset/www/#/status)

Obviously it wants to load the index file, ember is using hash for urls. This seems to cause an issue.

**Comment by MatthewPringle** [comment]
To elaborate on the issue...

What I think is happening, it that when the app is reloaded, ember simple auth is redirecting to the authenticated route, as it is supposed to if a valid token still exists.

But Cordova doesn't like that. I previously stopped Ember booting before Cordova was ready using a initializer and application.deferReadiness() until Cordova fired the 'deviceready' event.

That stopped a lot of bugs, but somehow I missed the above Android one.

To stop the network error from happening at all I changed locationType: 'none'

Now the url is not changed at all, the app no longer complains about network errors and ember simple auth can reroute on app boot.

The Android back button no longer works, but that can be fixed in code.

Not sure using 'hash' as location type is the best idea for Cordova apps as it has caused me issues on a good few occasions. It seems Cordova doesn't like anything appending to index.html, hashes etc...

Any ideas anyone?

**Comment by alexblom** [comment]
Do you have locationType: none set in your ember settings? That looks like it could be the culprit.

**Comment by alexblom** [comment]
In general there shouldn't be an issue with hashes in Cordova apps - hash routing won't be loading the index.html page anything past the first instance. It's interesting here that none resolves the issue for you. 

Is there perhaps something hardcoded in your app to force a window.reload or something similar?

**Comment by MatthewPringle** [comment]
From what I could gather online Cordova doesn't like anything appended to index.html on load.

So when the app tries to reload / recover and the url is file:///android_asset/www/#/status it fails. Status being a page in the app I have made.

I stopped any internal routing and the whole ember app waits for Cordova to have fully loaded before it attempts to do anything, using an initialiser. This was after a bug causes with Ember Simple Auth where it would automatically route a logged in user away from the index page on launch.
